apps/openmw/mwdialogue/dialoguemanagerimp.cpp

```diff
@@ -401,15 +400,18 @@ namespace MWDialogue
                     MWScript::InterpreterContext interpreterContext(&mActor.getRefData().getLocals(),mActor);
                     callback->addResponse("", Interpreter::fixDefinesDialog(text, interpreterContext));
 
-                    // Make sure the returned DialInfo is from the Dialogue we supplied. If could also be from the Info refusal group,
-                    // in which case it should not be added to the journal.
-                    for (ESM::Dialogue::InfoContainer::const_iterator iter = dialogue->mInfo.begin();
-                        iter!=dialogue->mInfo.end(); ++iter)
+                    if (dialogue->mType == ESM::Dialogue::Topic)
                     {
-                        if (iter->mId == info->mId)
+                        // Make sure the returned DialInfo is from the Dialogue we supplied. If could also be from the Greeting group,
+                        // in which case it should not be added to the journal.
```

Comment:
This comment should not be changed - the loop below it is still there to filter out Info Refusal topics.

Reply:
If you say so...